core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/puppeteer-utils.ts

```diff
@@ -126,54 +160,50 @@ module.exports = class baseUser {
 
   /**
    * This function types the text in the input field using its CSS selector.
-   * @param {string} selector - The CSS selector of the input field.
-   * @param {string} text - The text to be typed in the input field.
    */
-  async type(selector, text) {
+  async type(selector: string, text: string): Promise<void> {
     await this.page.waitForSelector(selector);
     await this.page.type(selector, text);
   }
 
   /**
    * This selects a value in a dropdown.
-   * @param {string} selector - The CSS selector of the input field.
-   * @param {string} option - The option to be selected.
    */
-  async select(selector, option) {
+  async select(selector: string, option: string): Promise<void> {
     await this.page.waitForSelector(selector);
     await this.page.select(selector, option);
   }
 
   /**
    * This function navigates to the given URL.
-   * @param {string} url - The URL to which the page has to be navigated.
    */
-  async goto(url) {
+  async goto(url: string): Promise<void> {
     await this.page.goto(url, {waitUntil: 'networkidle0'});
   }
 
   /**
    * This function uploads a file using the given file path.
-   * @param {string} filePath - The path of the file to be uploaded.
    */
-  async uploadFile(filePath) {
+  async uploadFile(filePath: string): Promise<void> {
     const inputUploadHandle = await this.page.$('input[type=file]');
     let fileToUpload = filePath;
-    inputUploadHandle.uploadFile(fileToUpload);
+    inputUploadHandle?.uploadFile(fileToUpload);
```

Comment:
What happens here if there's no input upload handle? Will the function fail? (It should...)

If not then maybe we should assert that the inputUploadHandle exists rather than letting it be potentially undefined?

Reply:
Done